pkg/saga/statemachine/engine/serializer/serializer.go

```diff
@@ -0,0 +1,47 @@
+package serializer
+
+import (
+	"bytes"
+	"encoding/gob"
+	"encoding/json"
+	"github.com/pkg/errors"
+)
+
+type Serializer[S any, T any] interface {
```

Comment:
Is this using generics? It's generally better not to use generics for now? @wt-better

Reply:
Yes, `Serializer` is a generic interface. This interface refers to the Java implementation (using generics as well). Is any specific reason that we should not use generics?

Reply:
I got it. There is no need to use generics. Fix it later.